### PR TITLE
#159987402 implement password reset

### PR DESCRIPTION
**Pull request by tomiadebanjo**
#### What does this PR do?
Implement password reset

#### Description of Task to be completed?
- Install sendgridAPI
- create endpoints for users to request for a password reset.
- setup functionality to allow users reset their password.

#### How should this be manually tested?
1. Clone the repository.
2. Checkout the `ft-user-password-reset-159987402` branch.
3. Run npm install on your local machine.
4. Run the db migrations using `npm run migrate`.
5. Start the server using `npm run start_dev`.
5. You can test the following endpoints using postman
`localhost:8000/api/v1/users/api/auth/reset-password/` 
`localhost:8000/api/v1/users/api/auth/reset-password/:token`

#### What are the relevant pivotal tracker stories?
[ #159987402](https://www.pivotaltracker.com/story/show/159987402)

### CODE REVIEW

File: server/controllers/userController.js [hunk 1]
```diff
@@ -66,6 +75,58 @@ const userController = {
...
+   * @param  {object} req The request object
+   * @param  {object} res The response object
+   * @returns {object} json response
+   */
+  resetPassword: (req, res) => {
+    Users
+      .findOne({
+        where: { email: req.body.email }
+      })
+      .then((user) => {
```
Comment (iAmao) [inline server/controllers/userController.js:104]: No catch block for this. If something happens users will never get a response

File: server/controllers/userController.js [hunk 2]
```diff
@@ -66,6 +75,58 @@ const userController = {
...
+  /**
+   * @description This is method that resets the users password
+   * @param  {object} req The request object
+   * @param  {object} res The response object
+   * @returns {object} json response
+   */
+  reset: (req, res) => {
+    Users.findOne({
+      where: { id: req.currentUser.id }
+    }).then((user) => {
```
Comment (iAmao) [inline server/controllers/userController.js:134]: No catch block for this. If something happens users will never get a response

File: server/controllers/userController.js [hunk 3]
```diff
@@ -66,6 +75,58 @@ const userController = {
...
+      if (!helpers.compareDate(req.currentUser.updatedAt, user.updatedAt)) {
+        return res.status(401).jsend.error({ message: 'Verification link not valid' });
+      }
+      Users.update({
+        password: bcrypt.hashSync(req.body.password, 8)
+      }, {
+        where: { id: req.currentUser.id }
+      }).then(() => res.status(200).jsend.success({
```
Comment (iAmao) [inline server/controllers/userController.js:142]: No catch block for this. If something happens users will never get a response

File: server/helpers/generateToken.js [hunk 4]
```diff
@@ -7,12 +7,12 @@ const secret = process.env.SECRET;
...
 /**
  * @description This function generates and encrypts JWT token
- * @param  {integer} id
  * @param  {integer} time
+ * @param  {object} object
  * @returns {string} encrypted JWT token
  */
-const generateToken = (id, time) => {
-  const rawToken = jwt.sign({ id }, secret, { expiresIn: time });
+const generateToken = (time, ...args) => {
```
Comment (iAmao) [inline server/helpers/generateToken.js]: You could just use an object instead of a spread operator

**Review by iAmao — CHANGES_REQUESTED** [review]
Nice, but always handle edge cases and errors, keep function as simple as possible.

File: server/controllers/userController.js (hunk 1 above)
Comment (tomiadebanjo) [reply server/controllers/userController.js:104]: Fixed

File: server/controllers/userController.js (hunk 2 above)
Comment (tomiadebanjo) [reply server/controllers/userController.js:134]: Fixed

File: server/controllers/userController.js (hunk 3 above)
Comment (tomiadebanjo) [reply server/controllers/userController.js:142]: I've added catch blocks for all the then's.

**Review by iAmao — APPROVED** [review]

**Pull request merged**